keygen/src/keygen.rs

```diff
@@ -281,6 +281,13 @@ fn app<'a>(num_threads: &'a str, crate_version: &'a str) -> Command<'a> {
                 .arg(
                     derivation_path_arg()
                 )
+                .arg(
+                    Arg::new("base58_keypair_string")
+                    .value_name("BASE58_KEYPAIR_STRING")
+                    .takes_value(true)
+                    .long("base58-keypair-string")
+                    .help("Base58 encoded keypair string")
+                )
```

Comment:
Is there a reason to include this in `new` rather than `recover`? `new` is meant to generate a new keypair, while `recover` is meant to do exactly what you're looking for: outputting a keypair file based on some keypair material.

I can't comment down there, but I think you could easily just extend the arg to `recover` to also `allow_base58()`, which will also require a new function on `SignerSourceParserBuilder`, and the rest should just work :tm: